Migrate mock cloud from iotile_analytics to python_iotile_cloud

timburke opened the pull request:
The mock cloud spins up a local http server and answers api requests
in a compatible manner to the real iotile cloud.  Two pytest fixtures
are registered for use in any package that depends on
python_iotile_cloud:

- mock_cloud
- mock_cloud_nossl

In order to use the first mock_cloud that serves over SSL (with a
self-signed certificate), you must also have pyOpenSSL installed.
See: requirements-test.txt

Closes #18

timburke committed:
Add support for testing using a mock iotile cloud

The mock cloud spins up a local http server and answers api requests
in a compatible manner to the real iotile cloud.  Two pytest fixtures
are registered for use in any package that depends on
python_iotile_cloud:

- mock_cloud
- mock_cloud_nossl

In order to use the first mock_cloud that serves over SSL (with a
self-signed certificate), you must also have pyOpenSSL installed.
See: requirements-test.txt

Closes #18

timburke committed:
Bump version to 0.7.4 for release and update changelog

timburke committed:
Increase appveyor test speed.

We already use tox to run tests against each python version,
specifying a matrix of python versions just changes what version
of python is used to invoke tox the first time, and for each
python version specified it still ran against all specified versions
in tox.

dkarchmer commented on CHANGELOG.md:
```diff
@@ -1,3 +1,8 @@
+### v0.7.4 (2018-01-26)
```
Probably big enough to go to v0.8.0

dkarchmer commented on setup.py:
```diff
@@ -14,6 +14,9 @@
         'iotile_cloud.stream'
     ],
+    entry_points={
+        'pytest11': ['iotile-analytics = iotile_cloud.utils.mock_cloud']
```
Is this line correct? `iotile-analytics`?

timburke replied:
sorry, shouldn't affect anything but yes an oversight.

dkarchmer commented on version.py:
```diff
@@ -1 +1 @@
-version = '0.7.3'
+version = '0.7.4'
```
v0.8.0

timburke replied:
understood.

dkarchmer reviewed (approved):
Let's call this v0.8.0. Thanks for doing the work.

timburke committed:
Fix copy-paste and update version to 0.8.0

timburke commented:
Also, I want to hold on releasing for a bit while I add in the features from https://github.com/iotile/python_iotile_cloud/issues/20 to this PR since the first place we use it is going to require them.

timburke committed:
Add ability to quickly mock device, project, streamer, org

See the functions quick_add_{device, project, org, streamer}
for descriptions of the functionality.  There is also a new
fixture mock_cloud_private[_nossl] that resets the cloud after
each test.

Closes #20

timburke committed:
Update fixtures for consistency

timburke committed:
Fix spelling

timburke committed:
Add support for listing device api with project filter

timburke committed:
Add support for including fleets in the mock cloud

timburke committed:
Remove debug print

timburke committed:
Fix python 3 compatibility of mock_cloud
